.github/workflows/scripts/install-go.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -e
+set -x
+
+GO_VERSION=1.22.0
+ARCH=`uname -i`
```

Comment:
Please use `uname -m` instead.  `-i` isn't supported on macOS and is documented as non-portable on ubuntu. As much as I can tell, they do the same thing.